jaeger-core/src/main/java/com/uber/jaeger/propagation/HexCodec.java

```diff
@@ -20,11 +20,13 @@
   /**
    * Parses a 1 to 32 character lower-hex string with no prefix into an unsigned long, tossing any
    * bits higher than 64.
+   *
+   * @return a 64 bit long, meaning that negative values are the overflow of Java's 32 bit long
    */
-  static long lowerHexToUnsignedLong(String lowerHex) {
+  static Long lowerHexToUnsignedLong(String lowerHex) {
```

Comment:
Generally, `Optional` would be more obvious. However I understand that some like to "optimise" code with `null`s

Reply:
I agree, this tradeoff was made to maintain Java 6 compatibility without adding/writing an optional back port.

Reply:
I get it. I didn't expect to have different lang level in `test` and `main`. Makes sense ;)